core/chains/evm/log/pool_unit_test.go

```diff
@@ -0,0 +1,343 @@
+package log_test
```

Comment:
Is using package `_test` necessary here? The usual case is to avoid an import cycle. We could reduce boilerplate by testing from inside `package log`.

Reply:
Testing from outside the package enforces the test to use only exported methods, and not any private methods.
Without that, it is easy for people to forget that, and directly test via private members.

Reply:
TBH I agree with Prashant here, I prefer to see all tests be in `XXX_test` package for two reasons:

1. Helps to avoid import cycles
2. If you're running tests with `./...` it sometimes doesn't work with tests that are defined in the same package (can't remember the exact circumstances but I've definitely run into this problem before)

If you need to test private functions, use `helpers_test.go` to export/

Reply:
@jmank88 @samsondav : Could we get a consensus here about whether we need in-package or _test package for unit-tests? Or we can get others' opinions too on a slack thread.

I like _test package because:
- It enforces contractually the use of interfaces. Avoid any accidental use of private fields.
- There is already a precedence of such use on internet, as well as in our codebase. Look at feeds manager unit-tests: https://github.com/smartcontractkit/chainlink/blob/8f366199b2d447febabec13bdb175c1c93b0f4d7/core/services/feeds/service_test.go
- Other things Sam said above.


Testing private functions can be done via exporting. I preferred separate exports_test.go just for all exports, but ok to put that all in helpers_test.go too, if people prefer that.

Reply:
@jmank88 has convinced me, it makes sense to put unit tests in the same package.